local_db/data_access.py

```diff
@@ -194,13 +194,18 @@ def delete_file_from_request(
         with transaction.atomic():
             # defense in depth
             request = self._find_metadata(request_id)
-            assert request.status == RequestStatus.PENDING
+
+            assert (
+                request.status == RequestStatus.PENDING
+                or request.status == RequestStatus.RETURNED
+            )
```

Comment:
This could use the new concept of status owner instead of listing specific statuses (`BusinessLogicLayer.STATUS_OWNERS[request.status] == RequestStatusOwner.AUTHOR`)

Reply:
done